src/test/ui/async-await/issues/issue-62517-2.rs

```diff
@@ -0,0 +1,16 @@
+// Regression test for #62517. We used to ICE when you had an `async
+// fn` with an `impl Trait` return that mentioned a `dyn Bar` with no
+// explicit lifetime bound.
+//
+// edition:2018
+
+#![feature(async_await)]
+
+trait Object {}
+
+trait Alpha<Param> {}
+
+async fn foo<'a>(_: &'a ()) -> impl Alpha<dyn Object> {}
+//~^ ERROR not satisfied
```

Comment:
On re-reading the PR... A small nit: Can we add the necessary impl to make this `// check-pass` instead?

Reply:
Done, pushed (at least I think it's done, didn't have time to let tests run to completion yet)